### ISSUE TITLE
✨ [REQUEST] Samba NAS: Enable multi-channel in smb.conf

### ISSUE THREAD

**Issue by codyrocco** [dianlight/hassio-addons#262]
### Addon

SambaNAS

### Checked

- [X] I have searched in the discussions & issues that it was not already covered

### Is your feature request related to a problem? Please describe

Giving the fact Samba NAS / HA runs nowadays on some really powerful minipcs, a lot of the,m having dual ethernet, why not enable multi-channel in Samba? Even a SATA SSD will exceed the capabilities of a 2.5Gbps link, then there's a lot of cheap 2.5Gbps switches...

### Describe the solution you'd like

Simply put in smb.conf these lines:

`server multi channel support = yes`
`aio read size = 1`
`aio write size = 1`

**Comment by dianlight** [comment]
I don't have a multi nic environment to test if it works. 
I will try to add the settings in the next beta version but I need a feedback before release in the next version. 
You are available for testing it?

L

**Comment by codyrocco** [comment]
sure I do. this could be done, anyway, with any nic available - for example, i did this not only with two pcie nic, but one pcie and one usb, or pcie wired + pcie wireless for the minipc,, and  one 10 gbpc nic with receive side scaling  at the test pc (or also two nic]
https://learn.microsoft.com/en-us/azure-stack/hci/manage/manage-smb-multichannel

**Comment by codyrocco** [comment]
Hmm... How is a docker container dealing with two nic? Must be host, both of them... this is beyond my knowledge, unfortunately.

**Comment by dianlight** [comment]
In the beta channel the new version has the option to enable multi-channel.
I try with my RPi4 with 2.5Gbps LAN (HW clapped it at 1Gb/s) + WLAN 250Mbs and I don't have any speed improvement also because my USB3 disk max read speed is 60Mbs. (I don't remember if is a disk or rpi4 limitation! ) 

In any case, multi-channel is working. I see the traffic on both NICs and if during the file transfer I physically unplug the cable or turn off the Wifi it does not interrupt but slows down for a few seconds and then resumes.

L.

**Comment by codyrocco** [comment]
That's enough. Great work, thank you!

**Comment by codyrocco** [comment]
Not wanting to open another issue - do you think there's a method to safe unmount a drive? It seems that's not enough to delete it from system/storage.

**Comment by dianlight** [comment]
In stable versions there is a bug.  So you need to stop the addon and reboot the host.
In beta version simply stop the addon.

L

**Comment by thcuba** [comment]
sometimes my ssd in is lost and in will "rebecome" present after a while.
i hope you could understand my bad english because i'm italian too
thank for your hard work

**Comment by dianlight** [comment]
Is not about English (my is miles away from decent ) but about netiquette. Put a question in a closed thread about another subject is not correct. 
I suggest you to open a new issue and because I don't have a magic ball add some information about your settings and environments.

L.

**Comment by thcuba** [comment]
ho usato il tuo samba nas e l'errore che avevo trovato riguardo a velocità
e montaggio non riguardava il tuo add on ma un ssd fallato.
mi piacerebbe testare il tuo srat che da quanto vedo è una versione
"nuova"... purtroppo stò imparando linux e il suo mondo e non ho la più
pallida idea di come installarlo in haos senza avere l'addon direttamente
scaricabile dal tuo repository.
grazie per il tuo lavoro

Il giorno ven 21 mar 2025 alle ore 05:52 Lucio Tarantino <
***@***.***> ha scritto:
